### PR TITLE
Add light on_state trigger

### PR DESCRIPTION
**Pull request by tony-fav**
## Description:

docs for `light.on_state` trigger

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2868

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/light/index.rst [hunk 1]
```diff
@@ -382,11 +384,12 @@ that the light is completely OFF, and ON means that the light is emitting at lea
 
 .. _light-on_turn_on_off_trigger:
 
-``light.on_turn_on`` / ``light.on_turn_off`` Trigger
-****************************************************
+``light.on_turn_on`` / ``light.on_turn_off`` / ``light.on_state`` Trigger
```
Comment (jesserockz) [inline components/light/index.rst]: Maybe make a new heading for `on_state`, it seems weird the way the paragraph works.

Comment (tony-fav) [reply components/light/index.rst]: Makes sense. I split it off into its own header now.

File: components/light/index.rst [hunk 2]
```diff
@@ -382,8 +384,8 @@ that the light is completely OFF, and ON means that the light is emitting at lea
 
 .. _light-on_turn_on_off_trigger:
 
-``light.on_turn_on`` / ``light.on_turn_off`` Trigger
-****************************************************
+``light.on_turn_on`` / ``light.on_turn_off``
+********************************************
```
Comment (oxan) [inline components/light/index.rst]: No need for this change anymore:
```suggestion
``light.on_turn_on`` / ``light.on_turn_off`` Trigger
****************************************************
```

File: components/light/index.rst [hunk 3]
```diff
@@ -398,6 +400,25 @@ with the behavior of the ``light.is_on`` and ``light.is_off`` condition above.
         on_turn_off:
         - logger.log: "Light Turned Off!"
 
+.. _light-on_state_trigger:
+
+``light.on_state`` Trigger
+**************************
+
+This trigger is activated each time the light set state is changed. It is not triggered
```
Comment (oxan) [inline components/light/index.rst]: I think this is more clear:
```suggestion
This trigger is activated each time the set light state is changed. It is not triggered 
```

**Review by oxan — CHANGES_REQUESTED** [review]

**Review by oxan — APPROVED** [review]

**Pull request merged**